third_party/move/move-model/src/metadata.rs

```diff
@@ -13,6 +13,8 @@ use std::{
 
 const UNSTABLE_MARKER: &str = "-unstable";
 
+pub static COMPILATION_METADATA_KEY: &[u8] = "compilation_metadata".as_bytes();
```

Comment:
I feel like it would be more consistent to move this into `move-binery-format`. My reasoning is that fundamental concepts like this are tied to the definition of the file format itself, rather than `move-model`, which is a higher level representation of the file format. In more practical terms, there may be tools that want to make sense of this metadata without wanting to depend on the move-model.

Same for `struct CompilationMetadata`. Actually maybe this whole file should be move into `move-binary-format`

What do you think @rahxephon89 @wrwg?

Reply:
@wrwg, if we want to attach the compilation metadata to the bytecode generated by V1 as well (which I have not done in this PR yet), we need to move `metadata.rs` out of `move-model` to avoid cyclic dependency issue. What do you think?